common/ast/src/parser/statement.rs

```diff
@@ -1177,16 +1199,23 @@ pub fn show_limit(i: Input) -> IResult<ShowLimit> {
     )(i)
 }
 
-pub fn table_option(i: Input) -> IResult<TableOption> {
-    alt((
-        map(engine, TableOption::Engine),
-        map(
-            rule! {
-                COMMENT ~ ^"=" ~ #literal_string
-            },
-            |(_, _, comment)| TableOption::Comment(comment),
-        ),
-    ))(i)
+pub fn table_option(i: Input) -> IResult<BTreeMap<String, String>> {
+    let opt_key = map_res(ident, |ident| {
+        if ident.quote.is_none() && ident.name != "AS" {
```

Comment:
`AS` should be a reserved keyword and is impossible to be an identifier unless it's parsed using `ident_after_as`.